### ISSUE TITLE
tests/ml/test_trainer.py test_create_engine() is flaky

### ISSUE THREAD

**Issue by ethman** [nussl/nussl#176]
Test linked here: https://github.com/nussl/nussl/blob/acb62b6cd29b5fdd08539bae905d2df1d83d62a0/tests/ml/test_trainer.py#L19

This test loads the best model after training for 3 epochs on a small dataset. It tests to make sure the state_dict of a new trainer matches that of the one used to train for those 3 epochs. However, the loss function is random and therefore the best model saver might not save all 3 epochs, which leads to a flaky test.
